mono/metadata/threads.c

```diff
@@ -2194,8 +2194,14 @@ void mono_thread_current_check_pending_interrupt ()
 void
 ves_icall_System_Threading_Thread_Abort (MonoInternalThread *thread, MonoObject *state)
 {
+	MonoThread *sys_thread = mono_thread_current ();
+
 	LOCK_THREAD (thread);
 	
+	if (sys_thread->pending_exception) {
```

Comment:
Looks like what we need for this change are some tests. And run them on windows.